casanovo/denovo/model.py

```diff
@@ -195,7 +196,18 @@ def forward(
             spectra.to(self.encoder.device),
             precursors.to(self.decoder.device),
         )
-        return [self.decoder.detokenize(t) for t in tokens], aa_scores
+        peptides = []
```

Comment:
**issue:** I don't understand this change. This copies the `detokenize` code from depthcharge and does exactly the same as what the previous one-line statement did, except that it's a list rather than a string.

1. Why is this change necessary? Both list and strings are iterables.
2. Even if this change is necessary, duplicating 95% of functionality from depthcharge is not the best solution.

Reply:
Regex logic in `peptide_mass_calculator.mass` downstream can't handle cases where multiple N-terminal modifications are predicted back to back (e.g. -17.027+42.011) if the peptide is `str` rather than list so it's useful to decode this way to have an AA list. We can implement a new regex to convert `str` to list here or `peptide_mass_calculator.mass` can be modified in DepthCharge .

Reply:
I prefer to keep the (de)tokenizing functionality fully on the depthcharge side, rather than mixing it in the different codebases. I suggest to revert the changes here and directly open a PR in depthcharge that will need to be integrated prior to merging this PR.

Reply:
Reverted to `detokenize` and updated depthcharge dependency